--- OgreCEGUIRenderer.cpp	Wed Jan 28 21:14:09 2009
+++ OgreCEGUIRenderer.cpp	Wed Jan 28 21:07:10 2009
@@ -24,11 +24,11 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 *************************************************************************/
 
-#include <CEGUI/CEGUIImagesetManager.h>
-#include <CEGUI/CEGUIImageset.h>
-#include <CEGUI/CEGUIImage.h>
-#include <CEGUI/CEGUIExceptions.h>
-#include <CEGUI/CEGUISystem.h>
+#include <CEGUIImagesetManager.h>
+#include <CEGUIImageset.h>
+#include <CEGUIImage.h>
+#include <CEGUIExceptions.h>
+#include <CEGUISystem.h>
 
 #include "OgreCEGUIRenderer.h"
 #include "OgreCEGUITexture.h"
@@ -454,7 +454,11 @@
 	d_render_sys->_setTextureUnitFiltering(0, FO_LINEAR, FO_LINEAR, FO_POINT);
 	d_render_sys->_setTextureAddressingMode(0, d_uvwAddressMode);
 	d_render_sys->_setTextureMatrix(0, Matrix4::IDENTITY);
+#if OGRE_VERSION >= 0x010600
+	d_render_sys->_setAlphaRejectSettings(CMPF_ALWAYS_PASS, 0, false);
+#else
 	d_render_sys->_setAlphaRejectSettings(CMPF_ALWAYS_PASS, 0);
+#endif
 	d_render_sys->_setTextureBlendMode(0, d_colourBlendMode);
 	d_render_sys->_setTextureBlendMode(0, d_alphaBlendMode);
 	d_render_sys->_disableTextureUnitsFrom(1);
--- OgreCEGUIRenderer.h	Wed Jan 28 21:14:09 2009
+++ OgreCEGUIRenderer.h	Wed Jan 28 21:06:46 2009
@@ -29,9 +29,9 @@
 #ifndef _OgreCEGUIRenderer_h_
 #define _OgreCEGUIRenderer_h_
 
-#include <CEGUI/CEGUIBase.h>
-#include <CEGUI/CEGUIRenderer.h>
-#include <CEGUI/CEGUITexture.h>
+#include <CEGUIBase.h>
+#include <CEGUIRenderer.h>
+#include <CEGUITexture.h>
 
 #include <OgreRenderQueueListener.h>
 #include <OgreSceneManagerEnumerator.h>
--- OgreCEGUIResourceProvider.cpp	Wed Jan 28 21:14:09 2009
+++ OgreCEGUIResourceProvider.cpp	Wed Jan 28 21:07:15 2009
@@ -25,7 +25,7 @@
 *************************************************************************/
 #include "OgreCEGUIResourceProvider.h"
 
-#include <CEGUI/CEGUIExceptions.h>
+#include <CEGUIExceptions.h>
 #include <OgreArchiveManager.h>
 
 
--- OgreCEGUIResourceProvider.h	Wed Jan 28 21:14:09 2009
+++ OgreCEGUIResourceProvider.h	Wed Jan 28 21:06:51 2009
@@ -26,7 +26,7 @@
 #ifndef _OgreCEGUIResourceProvider_h_
 #define _OgreCEGUIResourceProvider_h_
 
-#include <CEGUI/CEGUIResourceProvider.h>
+#include <CEGUIResourceProvider.h>
 #include "OgreCEGUIRenderer.h"
 
 // Start of CEGUI namespace section
--- OgreCEGUITexture.cpp	Wed Jan 28 21:14:09 2009
+++ OgreCEGUITexture.cpp	Wed Jan 28 21:07:21 2009
@@ -24,8 +24,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 *************************************************************************/
 
-#include <CEGUI/CEGUISystem.h>
-#include <CEGUI/CEGUIExceptions.h>
+#include <CEGUISystem.h>
+#include <CEGUIExceptions.h>
 #include "OgreCEGUITexture.h"
 #include "OgreCEGUIRenderer.h"
 
--- OgreCEGUITexture.h	Wed Jan 28 21:14:09 2009
+++ OgreCEGUITexture.h	Wed Jan 28 21:07:06 2009
@@ -26,9 +26,9 @@
 #ifndef _OgreCEGUITexture_h_
 #define _OgreCEGUITexture_h_
 
-#include <CEGUI/CEGUIBase.h>
-#include <CEGUI/CEGUIRenderer.h>
-#include <CEGUI/CEGUITexture.h>
+#include <CEGUIBase.h>
+#include <CEGUIRenderer.h>
+#include <CEGUITexture.h>
 #include "OgreCEGUIRenderer.h"
 
 // Start of CEGUI namespace section
