--- OISPrereqs.h	(revision 7141)
+++ OISPrereqs.h	(working copy)
@@ -60,7 +60,7 @@
 #		define OIS_XBOX_PLATFORM
 #	else
 #		define OIS_WIN32_PLATFORM
-#		if defined( OIS_DYNAMIC_LIB )
+#		if !defined( OIS_STATIC_LIB )
 #			undef _OISExport
 			//Ignorable Dll interface warning...
 #           if !defined(OIS_MINGW_COMPILER)


--- linux/EventHelpers.cpp	(revision 5668)
+++ linux/EventHelpers.cpp	(working copy)
@@ -35,6 +35,20 @@
 # include <iostream>
 #endif
 
+// Fixes for missing macros in input.h
+#ifndef FF_EFFECT_MIN
+#define FF_EFFECT_MIN FF_RUMBLE
+#endif
+#ifndef FF_EFFECT_MAX
+#define FF_EFFECT_MAX FF_RAMP
+#endif
+#ifndef FF_WAVEFORM_MIN
+#define FF_WAVEFORM_MIN FF_SQUARE
+#endif
+#ifndef FF_WAVEFORM_MAX
+#define FF_WAVEFORM_MAX FF_CUSTOM
+#endif
+
 using namespace std;
 using namespace OIS;
 

--- win32/Win32ForceFeedback.cpp
+++ win32/Win32ForceFeedback.cpp
@@ -25,7 +25,7 @@
 #include <math.h>
 
 // 0 = No trace; 1 = Important traces; 2 = Debug traces
-#define OIS_WIN32_JOYFF_DEBUG 1
+#define OIS_WIN32_JOYFF_DEBUG 0
 
 #if (defined (_DEBUG) || defined(OIS_WIN32_JOYFF_DEBUG))
   #include <iostream>


--- mac/MacHIDManager.cpp
+++ mac/MacHIDManager.cpp
@@ -406,6 +406,7 @@
 			switch(iType)
 			{
 				case OISJoyStick:
+                {
 					int totalDevs = totalDevices(iType);
 					int freeDevs = freeDevices(iType);
 					int devID = totalDevs - freeDevs;
@@ -413,6 +414,7 @@
 					obj = new MacJoyStick((*it)->combinedKey, bufferMode, *it, creator, devID);
 					(*it)->inUse = true;
 					return obj;
+                }
 				case OISTablet:
 					//Create MacTablet
 					break;
