--- linux/EventHelpers.cpp	(revision 5668)
+++ linux/EventHelpers.cpp	(working copy)
@@ -35,6 +35,20 @@
 # include <iostream>
 #endif
 
+// Fixes for missing macros in input.h
+#ifndef FF_EFFECT_MIN
+#define FF_EFFECT_MIN FF_RUMBLE
+#endif
+#ifndef FF_EFFECT_MAX
+#define FF_EFFECT_MAX FF_RAMP
+#endif
+#ifndef FF_WAVEFORM_MIN
+#define FF_WAVEFORM_MIN FF_SQUARE
+#endif
+#ifndef FF_WAVEFORM_MAX
+#define FF_WAVEFORM_MAX FF_CUSTOM
+#endif
+
 using namespace std;
 using namespace OIS;
 

--- win32/Win32ForceFeedback.cpp
+++ win32/Win32ForceFeedback.cpp
@@ -25,7 +25,7 @@
 #include <Math.h>
 
 // 0 = No trace; 1 = Important traces; 2 = Debug traces
-#define OIS_WIN32_JOYFF_DEBUG 1
+#define OIS_WIN32_JOYFF_DEBUG 0
 
 #if (defined (_DEBUG) || defined(OIS_WIN32_JOYFF_DEBUG))
   #include <iostream>
